lib/broccoli/ember-app.js

```diff
@@ -328,6 +328,25 @@ class EmberApp {
 
   /**
     @private
+    @method _needsShims
+  */
+  _needsShims(subject) {
```

Comment:
The extracted function is great, I think it would work a bit better as a separate stand alone function (e.g. not as a method on `EmberApp` itself), since we are using it unbound in the `addons.some(...)` bit just below.  As it stands, I'm not sure that more than two levels would work?

Reply:
aight, I have an idea for a different abstraction that I think is more flexible.

Reply:
@rwjblue updated!